### PR TITLE
Update SplitButton border to render like Button border

### LINKED ISSUE #6638: SplitButton border is different from Button border in dark mode
### Describe the bug

The border of the `SplitButton` is different from the one of the `Button` in dark mode.

### Steps to reproduce the bug

1. Open Xaml control gallery.
2. Set the app theme to dark
3. Open the `Button` sample page.
4. Open the `SplitButton` sample page.

Compare the border of the two buttons

### Expected behavior

The `SplitButton` border is different from the one on `Button`. It seems that linear gradient is not applied in dark mode. We can see it in light mode.

### Screenshots

image

### NuGet package version

Microsoft.UI.Xaml 2.7.0

### Windows app type

- [X] UWP
- [ ] Win32

### Device form factor

Desktop

### Windows version

Insider Build (xxxxx)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by chingucoding**
## Description

The root cause of the border of SplitButton not looking like the Button border is the fact that the border was rendered on top of the semi-light gray background of the control instead of being rendered against the background of the control surrounding the SplitButton. This caused the border to look differently in dark theme compared to the Button border.

## Motivation and Context

Closes #6638
## How Has This Been Tested?

Tested manually.
## Screenshots (if appropriate):

Before:
Screenshot of different Visual States of SplitButton control with faulty border

After:
Screenshot of different Visual States of SplitButton control with correct border rendering

### CODE REVIEW

**Comment by ojhad** [comment]
The 'CheckedTouchPressed' state in the after screenshot has square corners

File: dev/SplitButton/SplitButton.xaml [hunk 1]
```diff
@@ -160,9 +162,12 @@
                                     <VisualState.Setters>
                                         <Setter Target="PrimaryBackgroundGrid.Background" Value="{ThemeResource SplitButtonBackgroundChecked}"/>
                                         <Setter Target="SecondaryBackgroundGrid.Background" Value="{ThemeResource SplitButtonBackgroundChecked}"/>
-                                        <Setter Target="Border.BorderBrush" Value="{ThemeResource SplitButtonBorderBrushChecked}"/>
+                                        <Setter Target="RootGrid.BorderBrush" Value="{ThemeResource SplitButtonBorderBrushChecked}"/>
+                                        <Setter Target="RootGrid.BorderThickness" Value="0"/>
```
Comment (StephenLPeters) [inline dev/SplitButton/SplitButton.xaml:166]: This seems weird, why are we changing the color of a board with 0 thickness? and then setting the other button's border thickness to 1 without changing their bursh color?

Comment (chingucoding) [reply dev/SplitButton/SplitButton.xaml:165]: Regarding the first point, you are right; I removed the borderbrush setter. Regarding the second point, we are artificially increasing the buttons size by 1 px in all dimensions to deal with the fact that the outer border shrinked. Without this, we would have the button change in size if you would check/uncheck it.

Comment (dansmor7) [reply dev/SplitButton/SplitButton.xaml:165]: Why doesn't the colored button have a border anymore? It's supposed to have one, according to the design specs.

**Comment by dansmor7** [comment]
image
There is a little issue with how the corners look. How about applying the border to both PrimaryBackgroundGrid and SecondaryBackgroundGrid? I think that will need converters, so that it keeps working with the border thickness api.

File: dev/SplitButton/SplitButton.xaml (hunk 1 above)
Comment (chingucoding) [reply dev/SplitButton/SplitButton.xaml:165]: This is indeed wrong. The initial reason I removed it was that the control does not properly render when checked if we leave the border:

Screenshot of SplitButton in checked state rendering faulty

I'll have to dig into this again.

**Comment by chingucoding** [comment]
I wanted to avoid using converters but it seems that the border radius clipping does not work as expected. I'll dig into this.

**Comment by chingucoding** [comment]
I think the border should be fixed now without control parts sticking out.

**Comment by dansmor7** [comment]
Doesn't this make the border always have the same color? I think it's still supposed to be different for each button depending on the state.
image
image

**Comment by chingucoding** [comment]
I was able to separate them now with the use of converters, there is still a bug with clicking it and the border not updating correctly, however I am not sure if it is a bug with the template or UWP XAML since the visual states on its own all seem fine(ish).

**Comment by dansmor7** [comment]
Some states don't seem to update the border, and then a lot of brushes are identical.
There is one small issue with the divider. I think it should be contained within the border visually, rather than overlapping it:
image
image

Also, why does BackgroundSizing change for some states?

**Comment by chingucoding** [comment]
> Some states don't seem to update the border, and then a lot of brushes are identical.
Is this present with the latest changes and if so, can you elaborate on that?

> There is one small issue with the divider. I think it should be contained within the border visually, rather than overlapping it:
Fixed now.

> Also, why does BackgroundSizing change for some states?
This is how the AccentButton style and ToggleButton are implementing the border rendering with accent colored background. Current state does not use it anymore though.

**Comment by dansmor7** [comment]
Did the implementation with BackgroundSizing not work well? I think it was ok, and it was also simpler.

Regarding the borders: PointerOver should stay the same, Pressed should update one of them, and then Disabled and FlyoutOpen should update both borders.
When checked, all borders should be updated in all states, otherwise it will use some unchecked brushes.

Then there is a problem in the theme resources: `SplitButtonBorderBrushPressed` is the same as the normal and pointer over brushes, but it should have the value of `ControlStrokeColorDefaultBrush`.

**Comment by dansmor7** [comment]
It's a bit strange. With the previous implementation, when going from unchecked to checked, BackgroundSizing will not update unless you press the button. Simply hovering won't change it.
Everything else about the states does change.

It could be a bug with BackgroundSizing, but maybe something happens with the Pressed state which doesn't happen with others.

**Comment by chingucoding** [comment]
I'm inclined to say that this is a bug with the BackgroundSizing property. There were different things I tried with it (e.g. use it on a Border or a Grid) but all of them did not really work as expected.

**Comment by dansmor7** [comment]
I found the issue - #6885. I'm not sure if it can be fixed without WinUI 3, so this might be the best way to do it.
Interestingly enough, ToggleButton does not have this issue.

**Comment by chingucoding** [comment]
There could be a workaround but I don't think that there is a fix before WinUI 3. Maybe though, we are using Backgroundsizing completely wrong here.

**Comment by dansmor7** [comment]
Do you think the thickness filter could be added in a separate pr until this one is approved?
There are a lot of controls that would benefit from it, and it's kinda needed to fix some CommandBarFlyout issues.

**Comment by chingucoding** [comment]
> Do you think the thickness filter could be added in a separate pr until this one is approved?
> There are a lot of controls that would benefit from it, and it's kinda needed to fix some CommandBarFlyout issues.

We could try that but I am not sure if it would speed up the process much. In theory, first the filter would need to have the feature proposal process kicked off since it is a public API (@MikeHillberg FYI).

**Comment by gabbybilka** [comment]
Thank you for submitting this PR, the current focus of the team is to assist our users in moving from WinUI 2 to WinUI 3. We are closing this PR as it does not address resolving a critical issue for WinUI 2 or WinUI 3.

**Pull request closed without merging**